### PR TITLE
Make IFunction::executeImpl const

### PR DESCRIPTION
**Pull request by KochetovNicolai**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Just added ~500 times `const`.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
> Just added ~500 times const.

Ok. There also a few other modifications btw.

**Comment by alexey-milovidov** [comment]
Performance test Ok.

> Yandex third-party checks (only for Yandex employees) 

Arcadia has to be patched, @KochetovNicolai

**Comment by KochetovNicolai** [comment]
> Arcadia has to be patched

Done. Waiting for review from it.

**Pull request merged**
